Source/DafnyCore/Resolver.cs

```diff
@@ -10289,8 +10295,14 @@ public CheckDividedConstructorInit_Visitor(Resolver resolver)
 
         if (f.ByMethodBody != null) {
           var method = f.ByMethodDecl;
-          Contract.Assert(method != null); // this should have been filled in by now
-          ResolveMethod(method);
+          if (method != null) {
+            ResolveMethod(method);
+          } else {
+            // method should have been filled in by now,
+            // unless there was a function by method and a method of the same name
+            // but then this error must have been reported.
+            Contract.Assert(reporter.ErrorCount >= 1);
```

Comment:
nit: Why `>= 1` here but `> 0` in the other case?

Reply:
> nit: Why >= 1 here but > 0 in the other case?

i did not do them at the same time :-) will fix it for consistency.